php-practice/section12.php

```diff
@@ -159,3 +214,54 @@ class TwelveSidedDie extends Dice {
  * HINT: What shared methods or properties need to be in the parent HatOfNames class?
  * HINT: What visibility do these shared methods and properties need? (public, private, or protected?)
  */
+
+abstract class HatOfNames {
+  const MODE_RETAIN_WINNER = 1;
+  const MODE_NEW_WINNER_EVERY_TIME = 2;
+
+  protected $names = [];
+  protected $winner;
+  protected $mode;
```

Comment:
I'm thinking a phone or Skype call would be best for clarifying how to fix this. Just let me know.